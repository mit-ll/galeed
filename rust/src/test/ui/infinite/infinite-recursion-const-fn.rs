//https://github.com/rust-lang/rust/issues/31364

const fn a() -> usize { b() } //~ ERROR cycle detected when const-evaluating `a` [E0391]
const fn b() -> usize { a() }
const ARR: [i32; a()] = [5; 6];

fn main(){}
