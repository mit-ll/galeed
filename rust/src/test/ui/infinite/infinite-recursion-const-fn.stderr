error[E0391]: cycle detected when const-evaluating `a`
  --> $DIR/infinite-recursion-const-fn.rs:3:1
   |
LL | const fn a() -> usize { b() }
   | ^^^^^^^^^^^^^^^^^^^^^
   |
note: ...which requires const-evaluating `b`...
  --> $DIR/infinite-recursion-const-fn.rs:4:1
   |
LL | const fn b() -> usize { a() }
   | ^^^^^^^^^^^^^^^^^^^^^
   = note: ...which again requires const-evaluating `a`, completing the cycle
note: cycle used when const-evaluating `ARR::{{constant}}#0`
  --> $DIR/infinite-recursion-const-fn.rs:5:18
   |
LL | const ARR: [i32; a()] = [5; 6];
   |                  ^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0391`.
